app/components/admin/impersonate_component.rb

```diff
@@ -0,0 +1,10 @@
+module Admin
+  class ImpersonateComponent < ApplicationComponent
+    attr_reader :email
+
+    def initialize(email)
+      super
+      @email = email
+    end
+  end
+end
```

Comment:
I don't think we need a special component just for this. I also don't see how we would reuse this one as it's named a little specific.

I'd rather see something similar to `Admin::BasicLinkComponent` that takes in the form URL and an array of hidden fields. That way you could pass in the `:email` field. Also, the styling can match `Admin::BasicLinkComponent`.